aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -586,5 +587,30 @@ Tensor min(const Tensor& self, const Tensor& other) {
 
 Tensor& min_(Tensor& self, const Tensor& other) { return at::min_out(self, self, other); }
 
+bool equal_cpu(const Tensor& self, const Tensor& other) {
+  if (self.has_names() || other.has_names()) {
+    if (self.get_named_tensor_meta()->names() != other.get_named_tensor_meta()->names()) {
+      return false;
+    }
+  }
+  TORCH_CHECK(self.dtype() == other.dtype(),
+              "Expected object of scalar type ", self.dtype(), " but got scalar type ",
+              other.dtype(), " for argument 'other'");
+  if (!self.is_same_size(other)) {
+    return false;
+  }
+  std::atomic<bool> equal{true};
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND(kBool, self.scalar_type(), "equal_cpu", [&]() {
```

Comment:
Is it possible to move this part to `aten/src/ATen/native/cpu/BinaryOpsKernel.cpp`? There you can take advantage of SIMD instructions and the code structure is cleaner (especially after the CUDA version is also migrated). You can look into the other binary operators as examples.

Reply:
Thanks for the quick review.
I moved this to BinaryOpsKernel.